src/main/java/io/anserini/kg/freebase/EntityLinking.java

```diff
@@ -0,0 +1,360 @@
+        notfound += 1;
+      }
+    }
+
+    //Close the input stream
+    br.close();
+
+    LOG.info("Found: " + found);
+    LOG.info("Not Found: " + notfound);
+    LOG.info("Querying completed.");
+  }
+
+  public void searchGoldFile(String fileName) throws Exception {
+    LOG.info("fileName: " + fileName);
+    LOG.info("numHits: " + numHits);
+
+    // Open the file
+    FileInputStream fstream = new FileInputStream(fileName);
+    BufferedReader br = new BufferedReader(new InputStreamReader(fstream));
+
+    String strLine;
+
+    //Read File Line By Line
+    int found = 0, notfound = 0;
+    List<RankedEntity> rankedEntities;
+    while ((strLine = br.readLine()) != null)   {
+//      LOG.info("line: " + strLine);
+      String[] lineItems = strLine.split(" %%%% ");
+      String shortMid = getShortMid( cleanUri(lineItems[0].trim()) );
+      if (lineItems.length < 5) {
+        LOG.info(String.format("SHORT LINE < 5 items! line: %s", strLine));
+        continue;
+      }
+      String[] queries = lineItems[4].trim().split(" &&&& ");
+      boolean isFound = false;
+      int index = -1;
+      for (String query: queries) {
+        if (!isFound) {
+          query = removeQuotes(query.trim().toLowerCase());
+          try {
+            rankedEntities = search(query);
+          } catch (Exception e) {
+            // only keep the alphabets in the query
+            String queryStripped = query.replaceAll("[^a-zA-Z\\s]", "").replaceAll("\\s+", " ");
+            if (queryStripped.trim().length() > 0) {
+              rankedEntities = search(queryStripped);
+            } else {
+              LOG.info(String.format("WEIRD! query: %s", query, index));
+              continue;
+            }
+          }
+
+          RankedEntity entityMidToCompare = new RankedEntity(shortMid, 0.0f, "", "");
+          if (rankedEntities.contains(entityMidToCompare)) {
+            isFound = true;
+            index = rankedEntities.indexOf(entityMidToCompare);
+            break;
+          }
+        }
+      }
+      if (isFound) {
+        found += 1;
+        LOG.info(String.format("queries: %s,\tfound at index: %d", Arrays.asList(queries), index));
+      } else {
+        notfound += 1;
+        LOG.info(String.format("queries: %s,\tNOT found", Arrays.asList(queries), index));
+      }
+    }
+
+    //Close the input stream
+    br.close();
+
+    LOG.info("Found: " + found);
+    LOG.info("Not Found: " + notfound);
+    double percent = (found * 100.0) / (found + notfound);
+    LOG.info("Found/Total %: " + percent);
+    LOG.info("Querying completed.");
+  }
+
+  /**
+   * Prints query results to the standard output stream.
+   *
+   * @param queryName the entity name to search
+   * @throws Exception on error
+   */
+  public List<RankedEntity> search(String queryName) throws Exception {
+    // Initialize index searcher
+    IndexSearcher searcher = new IndexSearcher(reader);
+
+    Similarity similarity = new ClassicSimilarity();
```

Comment:
We may try out different similarity classes later so let's keep this here for now.